Include/Rover/LinearRegressionModel.hpp

```diff
@@ -43,19 +45,24 @@ namespace Rover {
       LinearRegressionModel<T>::operator ()(Arg&&... arg) const {
     auto x = dlib::matrix<Result, 1, std::tuple_size_v<
       typename Trial::Sample::Parameters> + 1>();
-    x = { 1., std::forward<Arg>(arg)... };
-    auto result = static_cast<Result>(x * m_transformation);
+    auto coef = std::vector<Result>();
```

Comment:
Linear regression is only defined for real numbers as both input and output variables. So each variable is supposed to represent a floating point type. std::move will handle explicit conversions as well as implicit ones.

Reply:
I can see why linear regression won't work for types like ```bool``` or ```std::string``` but I don't see why it wouldn't work for types representing prices, velocity, time durations, etc... Can you perhaps look into this more to see what approaches exist so that we can perform linear regression on types other than just ```double```? If a change to the API is needed, propose that, if other options exist feel free to investigate them, but it will be very user-unfriendly if we have to explicitly convert all data types that we use for backtesting trading strategies to ```double``` in order to use linear regression on it.

Reply:
LinearRegressionModel does not require every variable to be a double, it requires them to be convertible to doubles (or, technically, Result). All integers are naturally convertible to doubles or narrower types.
It may be more correct to hardcode the type (e.g. double) used for computation, and I actually made it that way initially. To be more generic, the class template can be parameterized with such a type to provide with a precision vs performance trade-off. In that case, Result also does not need to be an arithmetic type, but rather any type that can be initialized with a double or any narrower type.
Note that variables do not have to be directly convertible to doubles but any narrower types will also work. So, if prices, velocity etc. define conversions to an integer type (and I think they do in nexus), then they will work by virtue of type promotion.

There are techniques (e.g. factorization) to handle non-arithmetic types, but their utility is case-specific and therefore should not be incorporated into the regression itself. Rather, if a user wants to model a categorical variable using such techniques, it is up to them to select the proper one. Those are never exact; certain approaches will also increase the number of variables.

Reply:
Also, if I do not rely on type conversion, then there is no other protocol for me to understand that a type is numeric. So even though something like Price interfaces an integer, I will have to render it as a non-numeric type and apply one of those techniques to treat it like a categorical variable, which will certainly yield poor results.